job_creator.php

```diff
@@ -71,8 +68,61 @@ public function createJob(int $phpIndex, array $opts): array
         ];
         return array_merge($default, $opts);
     }
+
+    private function getPhpVersion(int $phpIndex): string
+    {
+        $key = str_replace('.x-dev', '', $this->installerVersion);
+        $repo = explode('/', $this->githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_LOCKSTEPPED_REPOS)) {
+            $cmsMajor = $this->getCmsMajorFromBranch();
+            $branch = $this->getCleanedBranch();
+            if (preg_match('#^[1-9]$#', $branch)) {
+                $key = $cmsMajor;
+            } elseif (preg_match('#^[1-9]\.([0-9]+)$#', $branch, $matches)) {
+                $key = sprintf('%d.%d', $cmsMajor, $matches[1]);
+            }
+        }
+        $phpVersions = INSTALLER_TO_PHP_VERSIONS[$key] ?? INSTALLER_TO_PHP_VERSIONS['4'];
+        return $phpVersions[$phpIndex] ?? $phpVersions[count($phpVersions) - 1];
+    }
+
+    private function getCmsMajorFromBranch(): string
+    {
+        $branch = $this->getCleanedBranch();
+        $repo = explode('/', $this->githubRepository)[1];
+        $branchMajor = '';
+        if (preg_match('#^[1-9]$#', $branch)) {
+            $branchMajor = $branch;
+        } elseif (preg_match('#^([1-9])\.[0-9]+$#', $branch, $matches)) {
```

Comment:
These won't work for major 10.x, but we're a long way off that so I'm not concerned about it. Just pointing it out as a matter of course.